packages/material-ui-lab/src/useAutocomplete/useAutocomplete.js

```diff
@@ -653,7 +653,9 @@ export default function useAutocomplete(props) {
     }
 
     if (autoSelect && selectedIndexRef.current !== -1) {
-      handleValue(event, filteredOptions[selectedIndexRef.current]);
+      selectNewValue(event, filteredOptions[selectedIndexRef.current]);
+    } else if (freeSolo && inputValue !== '') {
```

Comment:
@oliviertassinari Please keep in mind that I've removed `autoSolect`  from your  [previous example](https://github.com/mui-org/material-ui/issues/19270#issuecomment-575596764)

Reply:
Thanks, I will refactor a bit the tests, we should be good to go (keeping the auto-select).

Regarding changing the default auto select behavior. I don't remember exactly why I went with this value by default. We could explore it.